src/DetectorConstruction.cpp

```diff
@@ -93,17 +65,49 @@ G4VPhysicalVolume* DetectorConstruction::Construct()
   return world_phys_vol;
 }
 
+G4LogicalVolume* DetectorConstruction::BuildBoxDetector() {
+  G4Material* detector_mat = G4NistManager::Instance()->FindOrBuildMaterial("G4_lAr");
+
+  G4Box* detector_solid_vol = new G4Box("detector.solid", 
+                                        ConfigManager::GetDetectorWidth() / 2., 
+                                        ConfigManager::GetDetectorHeight() / 2., 
+                                        ConfigManager::GetDetectorLength() / 2.);
+
+  G4LogicalVolume* logic_vol = new G4LogicalVolume(detector_solid_vol, detector_mat, "detector.logical");
+  return logic_vol;
+}
+
+G4LogicalVolume* DetectorConstruction::BuildHDDetector() {
+  ConfigManager::SetDetectorWidth(2.3 * CLHEP::m);   // detector_x
+  ConfigManager::SetDetectorHeight(6.0 * CLHEP::m);  // detector_y
+  ConfigManager::SetDetectorLength(3.6 * CLHEP::m);  // detector_z
+  return BuildBoxDetector();
+}
+
+G4LogicalVolume* DetectorConstruction::BuildVDDetector() {
+  ConfigManager::SetDetectorWidth(20.0 * CLHEP::m);  // detector_x
+  ConfigManager::SetDetectorHeight(13.0 * CLHEP::m); // detector_y
+  ConfigManager::SetDetectorLength(6.5 * CLHEP::m);  // detector_z
+  return BuildBoxDetector();
+}
+
+G4LogicalVolume* DetectorConstruction::BuildTSDetector() {
+  ConfigManager::SetDetectorWidth(0.04 * CLHEP::m);  // detector_x
+  ConfigManager::SetDetectorHeight(0.04 * CLHEP::m); // detector_y
+  ConfigManager::SetDetectorLength(0.1 * CLHEP::m);  // detector_z
+  return BuildBoxDetector();
+}
+
 void DetectorConstruction::ConstructSDandField()
 {
   // SENSITIVE DETECTOR ////////////////////////////////////
-
   TrackingSD* tracking_sd = new TrackingSD("/G4QPIX/TRACKING", "TrackingHitsCollection");
   G4SDManager::GetSDMpointer()->AddNewDetector(tracking_sd);
 
-  G4LogicalVolume* detector_logic_vol =
-    G4LogicalVolumeStore::GetInstance()->GetVolume("detector.logical");
-
-  SetSensitiveDetector(detector_logic_vol, tracking_sd);
-
+  if (detector_logic_vol) {
```

Comment:
I would suggest moving the if-statement up to the beginning of the function, with something like
```
if (!detector_logic_vol) {
  G4cerr << "Error: detector_logic_vol is null in ConstructSDandField." << G4endl;
  return;
}

Then the rest of the function doesn't have to be wrapped up in an if block.

Reply:
Made the changes suggested!